Kernel/Bus/USB/UHCI/UHCIController.cpp

```diff
@@ -387,6 +389,37 @@ void UHCIController::dequeue_qh(QueueHead* transfer_queue)
     transfer_queue->prev_qh()->link_next_queue_head(transfer_queue->next_qh());
 }
 
+ErrorOr<QueueHead*> UHCIController::create_transfer_queue(Transfer& transfer)
+{
+    Pipe& pipe = transfer.pipe();
+
+    // Create a new descriptor chain
+    TransferDescriptor* last_data_descriptor;
+    TransferDescriptor* data_descriptor_chain;
+    auto buffer_address = Ptr32<u8>(transfer.buffer_physical().as_ptr());
+    TRY(create_chain(pipe, transfer.pipe().direction() == Pipe::Direction::In ? PacketID::IN : PacketID::OUT, buffer_address, pipe.max_packet_size(), transfer.transfer_data_size(), &data_descriptor_chain, &last_data_descriptor));
```

Comment:
Pre-existing since I see this code was just moved around, but passing two uninitialized pointers by address sets off my "This is C" alarm.

At the very least can we explicitly init those guys to nullptr? I would prefer if we could not use out-params like this and return `ErrorOr<SomeTinyStructWithTwoPointers>` rather than `ErrorOr<void>`

Reply:
Agreed it would be nice to refactor this.